src/preload/rrcalls.h

```diff
@@ -84,3 +84,8 @@
  * cases between the time slice signal and other rr behavior.
  */
 #define SYS_rrcall_arm_time_slice (RR_CALL_BASE + 10)
+/**
+ * Requests that rr not schedule a given thread. Used to test
+ * interactions with rr's scheduler
+ */
+#define SYS_rrcall_freeze_tid (RR_CALL_BASE + 11)
```

Comment:
Please document the parameters (tid) and exactly what this means. Is the thread never scheduled again?

Also, in many tests I've achieved similar results by making the "frozen" thread lower priority than other threads. Does that not work here for some reason?

Reply:
Will add documentation. The parameters are (tid, enable). rr will refuse to schedule the thread until called again with enable=0.

The reason for the rrcall is that I need different scheduling behavior between the kernel and rr's scheduler. I need the kernel to schedule it relatively soon (to wake it up while the signal is still pending and proceed to the exit stop), with enough headway for rr to notice that the thread was stopped, but I don't want rr to schedule it again until we've dequeued all the signals from the shared queue.